ocsp/responder.go

```diff
@@ -159,7 +171,20 @@ func (rs Responder) ServeHTTP(response http.ResponseWriter, request *http.Reques
 		return
 	}
 
+	parsedResponse, err := ocsp.ParseResponse(ocspResponse, nil)
+	if err != nil {
+		log.Errorf("Error parsing response: %s", err)
+		response.Write(unauthorizedErrorResponse)
+		return
+	}
+
 	// Write OCSP response to response
+	response.Header().Add("Last-Modified", parsedResponse.ProducedAt.Format(time.RFC1123))
+	response.Header().Add("Expires", parsedResponse.NextUpdate.Format(time.RFC1123))
+	maxAge := int64(parsedResponse.NextUpdate.Sub(rs.clk.Now()) / time.Second)
+	if maxAge > 0 {
+		response.Header().Add("Cache-Control", fmt.Sprintf("max-age=%d", maxAge))
```

Comment:
Oh, and `Last-Modified` should be `ThisUpdate` rather than `ProducedAt` per the same spec.